environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -199,6 +199,15 @@
      */
     public static final String DEV_MODE_SYSTEM_PROPERTY = "org.jboss.weld.development";
 
+    /**
+     * Standard behavior is that empty {@code beans.xml} is treated as discovery mode {@code annotated}.
+     * This configuration property allows to change the behavior to discovery mode {@code all} which is how it used to work prior to
+     * CDI 4.0.
+     * <p/>
+     * Note that this option is temporary and servers to easy migration. As such, it will be eventually removed.
+     */
+    public static final String LEGACY_EMPTY_BEANS_XML_TREATMENT = "org.jboss.weld.se.discovery.emptyBeansXmlModeAll";
```

Comment:
I like the name used in the key more, e.g. something like `EMPTY_BEANS_XML_DISCOVERY_MODE_ALL`?